### ISSUE TITLE
Preprocessor variable for commit number

### ISSUE THREAD

**Issue by jodavies** [form-dev/form#194]
If I understand correctly, the `347` in
```
TFORM 4.1 (Jun  6 2017, v4.1-20131025-347-g76561be) 64-bits
```
is an index that increments with commits?

Since there are regular bugfixes and feature additions in this repository it could be useful to have a way to check the form build is sufficiently up-to-date to run a script. Making such an index available as a preprocessor variable would be sufficient.

Or, is there already a way to do this?

Thanks,
Josh.

**Comment by tueda** [comment]
Basically it is just what `git describe --tags` returns and embedded by automake on Unix-like systems (but see also #119). It could be possible to add a preprocessor variable for that number, but this will not be available in builds without automake. And, Jos never uses automake, which means binaries sitting in the official FORM site will not get such a feature...

**Comment by jodavies** [comment]
This could also be implemented in a more coarse-grained way by incrementing the minor version number regularly. At the moment there is not even a minor version of form that will run forcer.

**Comment by tueda** [comment]
I agree with you that a more coarse-grained versioning is useful. Historically, I failed to persuade the developers to do that, which led to our versioning with the release date v4.0-20120330, v4.0-20120410, v4.1-20131025.
